### ISSUE TITLE
upload/model/checkout/marketing.php appears to be useless and should be deleted.

### ISSUE THREAD

**Issue by lucasjkr** [opencart/opencart#6134]
It's not called from anywhere. It has one method (getMarketingByCode) which is in another class
`upload/model/marketing/marketing.php`, and that class is called elsewhere.

This is probably just a leftover from an earlier time.

**Comment by zemguru** [comment]
Either it's left from OC2, or it can be used by some payment extensions?
`getMarketingByCode` found in these files on my OC2 version:

![image](https://user-images.githubusercontent.com/30476178/31962698-f0af791a-b906-11e7-80b2-8cb45ff55b1b.png)

**Comment by lucasjkr** [comment]
`getMarketingByCode()` is in the marketing/marketing.php file. And that's the model that is loaded when this method is called.

Compare:

File I suggest removing (last modified 3 years ago):
https://github.com/opencart/opencart/blob/master/upload/catalog/model/checkout/marketing.php

Marketing/marketing.php (last modified 1 month ago):
https://github.com/opencart/opencart/blob/master/upload/catalog/model/marketing/marketing.php

I did not find ANY controllers that use the `getMarketingByCode()` loading the one from checkout. 

Again, I think this file was probably used at one point, but got forgotten about.

**Comment by straightlight** [comment]
- I did not find ANY controllers that use the getMarketingByCode() loading the one from checkout.

Followed are the list of all files with mention of getMarketingByCode() method:

admin/controller/marketing/marketing.php
admin/model/marketing/marketing.php
catalog/controller/checkout/confirm.php
catalog/model/checkout/marketing.php
catalog/controller/extension/payment/amazon_login_pay.php
catalog/controller/extension/payment/klarna_checkout.php
catalog/controller/extension/payment/pilibaba.php
catalog/controller/extension/payment/pp_braintree.php
catalog/controller/extension/payment/pp_express.php

**Comment by lucasjkr** [comment]
But they all call it from the marketing.php class:

`$this->model_marketing_marketing->getMarketingByCode(...)`

**Comment by straightlight** [comment]
The admin-end calls the getMarketingCode() method from the marketing model class but the catalog-end calls the getMarketingCode() method from the checkout model class ... where is the problem exactly?

**Comment by lucasjkr** [comment]
what happens in Admin is irrelevant - completely independent system over there.

On Catalog side, in 3.0, getMarketingByCode is accessed using /upload/model/marketing/marketing.php class.

This file is never used:
/upload/catalog/model/checkout/marketing.php

Should be deleted. It's left over from 2.3

**Comment by straightlight** [comment]
I fail to see the understanding of this post since even the checkout controller is gathering the marketing code for validation purposes which means the marketingcode method is not being leftover anywhere from the list I provided above ... I even checked the TWIG files for any leftovers as all the relations in-between those calls seem to be accurate ... could you perhaps provide the line of code and the filename from where you believe the getMarketingCode() method would be used uselessly?

**Comment by lucasjkr** [comment]
You're not understanding this.

getMarketingByCode() is a method in TWO classes

One class file, located in `/upload/catalog/checkout/marketing.php` contains only that method and is NEVER USED anywhere.

The class file that also contains that method AND is used is in `/upload/catalog/marketing/marketing.php`

There is never any instance of $this->load->model('checkout/marketing'), nor is there ever $this->model_checkout_marketing->getMarketingByCode();

It's all done through marketing/marketing

This is a redundant file.

I don't know how else to explain it. But that file can be safely deleted and will make no impact on the site whatsoever because it's never used. The method that is in it is in another class.

**Comment by straightlight** [comment]
I do not know when you actually downloaded Opencart but in catalog/controller/checkout/confirm.php file ... see the line between 279 and 288 ... the model is being called and used as intended with the $this->model_checkout_marketing->getMarketingCode() for validation purposes ...

**Comment by lucasjkr** [comment]
I don't know why this is so difficult! Besides I don't think you even have commit privileges, this a simple thing for @danielkerr.

I don't care about the version you downloaded, only the version that is being developed here.

Go to the file you referred me to. Go to the lines that you referred me to. Now LOOK.

That line of code doesn't exist. It's using `model_marketing_marketing->` NOT `model_checkout_marketing->`

I've linked to the relevant files so many times, and then you go and look at the version you have on your server and not the ones I'm referring to.

<img width="1009" alt="screen shot 2017-10-25 at 11 08 27 am" src="https://user-images.githubusercontent.com/4703516/32006664-6121c73e-b975-11e7-9339-ccdd635328f6.png">

Now please go look at the code in this repository. Not your server.

**Comment by straightlight** [comment]
Yes but from which filename are you referring this screenshot from?

**Comment by straightlight** [comment]
From pilibaba.php file ... I have the checkout one... however, it should not be removed but rather replaced with:

$marketing_info = $this->model_checkout_marketing->getMarketingByCode($this->request->cookie['tracking']);

Since I found your enquiry in the code tab of Github as well, yes - confirmed - this would be a bug. :)

**Comment by lucasjkr** [comment]
I'm looking at the file that YOU told me to.

https://github.com/opencart/opencart/blob/master/upload/catalog/controller/checkout/confirm.php

it is not loading $this->model_checkout_marketing-> as you claimed.

Now I understand why Daniel gets such a bad rap, I really don't have any more patience. I've explained this a dozen times to you.

The same method is in two files.

In version 2.3 of Opencart, it used the method that is in File 1.

In version 3.0, it now uses the method in File 2.

That method is the only function in File 1. 

File 1 can be safely deleted.

Stop asking me anymore, just look at the code on this site, not an old version you downloaded. Not even a new version you downloaded. But this code itself, because this is what will make a future version.

**Comment by lucasjkr** [comment]
This is dumb.

You don't need to make any changes to any controllers.

Just delete an old file that's no longer being used.

Why isn't this coming through?

**Comment by straightlight** [comment]
The pilibaba.php file is a payment extension module that may still be in use by merchants even in the latest version of Opencart. However, I have now created a topic on the forum with these bug fixes: https://forum.opencart.com/viewtopic.php?f=201&t=198996

**Comment by lucasjkr** [comment]
One last time. 

RIGHT HERE ON GITHUB the PILIBABA extension is correct. It's getting the model from mode_marketing_marketing->getMarketingByCode.

No corrections or changes need to be made anywhere.

**Comment by straightlight** [comment]
That would be incorrect statement. If you look at this model file, the getMarketingByCode method is not implicit in the model class: https://github.com/opencart/opencart/blob/master/upload/catalog/model/marketing/marketing.php

In other words, I did properly looked into the github files.

**Comment by lucasjkr** [comment]
It becomes implicit when you load the model.

$this->load->model('marketing/marketing') only loads that one mode, not the checkout model.

And since this model has the method we need, and it's in a class whose name reflects its functionality, it's perfectly fine to leave it here.

There is never an instance of $this->load->model('checkout/marketing'), nor is there $this->model_checkout_marketing

In other words, the rest of the code base has already been refactored to account for this method being moved to a different class. All that remains is to delete the file. That's all I've been asking.

**Comment by straightlight** [comment]
I seem to have downloaded Opencart at a period when the marketing folder did not exist in the catalog/model for some reason ... however, the getMarketingByCode() method does seem to exist in both files. Which means, the checkout model and the marketing model are both calling the same method with the same table with the same SQL query ... perhaps it might be the one from the checkout that may no longer be in use ... Daniel will probably decide from there ...

**Comment by lucasjkr** [comment]
Glad we got that straightened out.

For future reference, when people make issues on Github, look at the code that's on Github at that moment, not the version you downloaded earlier. This was such an incredible waste of time and cause for huge frustration all because you refused to look at teh code I referred to and instead looked at your outdated code.

**Comment by straightlight** [comment]
Looking at the codes on Github, which I did, still does not explain the duplication of both statement rather on the checkout and on the marketing file with the same method name which brings the conclusion that we are both right and both wrong due to an earlier replacement and addition that was also duplicated on Github with those method names rather than simply moving that method name into a new folder and filename which even brought more confusion to the people.

**Comment by lucasjkr** [comment]
No, we're not both right. I'm correct. A file should be deleted. And you just wasted 2 hours of my life that i'll never get back.

**Comment by straightlight** [comment]
The one that may have to be deleted would be from catalog/model/checkout/marketing.php file. However, it would be hard to tell between the twos since there are no right and wrong here.

**Comment by lucasjkr** [comment]
That's what I said in the first place. It's not debatable - this file isn't being used, the other file is being used.

Will you be quiet now? You're not adding any value here.

**Comment by straightlight** [comment]
My apologize for the confusion. It seem to have been brought with much confusions regarding the use of two methods which I have now announced on the forum topic for users to download the new marketing model file to avoid future confusions as this issue was never reported before as you seem to talk about a filename while I thought it was about the method itself.

**Comment by lucasjkr** [comment]
I accept the apology.

But i will say I was clear this whole time that I was referring to a file and not a method.

Oh well. Lets move forward.

**Comment by straightlight** [comment]
Then it is settled ... since I do have the line of code that we no longer see on Github, in order to use Opencart, we need to download it everyday.

**Comment by lucasjkr** [comment]
Not in order to use opencart - the official releases are perfectly fine. But if you're developing OC or something related, then yes, you should get updated code regularly. Or just work on a fork, and then you only have to concern yourself with commits that affect the files you're working on.

Hey, would you mind sending me an email? It's on my profile page, just to take conversation off the Issues? Thanks!
